bookkeeper-server/pom.xml

```diff
@@ -284,16 +284,37 @@
             <exclude>**/.project</exclude>
             <exclude>**/.checkstyle</exclude>
             <exclude>**/.settings/*</exclude>
-            <exclude>certs/keyStoreClientPassword.txt</exclude>
-            <exclude>certs/keyStoreServerPassword.txt</exclude>
-            <exclude>certs/trustStorePassword.txt</exclude>
-            <exclude>src/test/resources/cacerts</exclude>
+            <exclude>src/test/resources/server-key.pem</exclude>
+            <exclude>src/test/resources/server-key.p12</exclude>
+            <exclude>src/test/resources/server-key.jks</exclude>
+            <exclude>src/test/resources/server-cert.pem</exclude>
+            <exclude>src/test/resources/client-key.pem</exclude>
+            <exclude>src/test/resources/client-key.p12</exclude>
+            <exclude>src/test/resources/client-key.jks</exclude>
+            <exclude>src/test/resources/client-cert.pem</exclude>
             <exclude>src/test/resources/keyStoreClientPassword.txt</exclude>
             <exclude>src/test/resources/keyStoreServerPassword.txt</exclude>
-            <exclude>src/test/resources/trustStorePassword.txt</exclude>
           </excludes>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>1.6.0</version>
+        <executions>
+            <execution>
+                <id>Generate Self-Signed Certificates</id>
+                <phase>generate-test-resources</phase>
+                <goals>
+                    <goal>exec</goal>
+                </goals>
+                <configuration>
+                    <workingDirectory>${basedir}/src/test/resources</workingDirectory>
+                    <executable>${basedir}/src/test/resources/generateKeysAndCerts.sh</executable>
```

Comment:
This is going to create new certificates at every build, two problems:
1) this is slow
2) it requires some jind of setup of openssl

I suggest to add a profile to build the certificates and commit a working version of certificates.
When they will expire we will do another commit

Reply:
How slow is this?

Reply:
With this patch at every "mvn test" we are going to re-create new certificates.
I think we have two options:
1) commit certificates and rebuild them when expired
2) make the script rebuild the certificate only if it does not exist

I prefer 1)as it makes tests reproducible at 100% and it does not impact build and dev time

Reply:
@eolivelli If it makes sense, I can commit certificates with a very long expiry so that we may never need to generate the certificates again. Also move certification generation to a new profile, just in case we need it.

Reply:
It would be fine for me, thank you

Reply:
+1 on @kishorekasi 's proposed approach.